tests/components/mqtt/test_config_flow.py

```diff
@@ -1060,6 +1061,92 @@ async def test_skipping_advanced_options(
     assert result["step_id"] == step_id
 
 
+@pytest.mark.parametrize(
+    ("test_input", "user_input", "new_password"),
+    [
+        (
+            {
+                mqtt.CONF_BROKER: "test-broker",
+                mqtt.CONF_USERNAME: "username",
+                mqtt.CONF_PASSWORD: "verysecret",
+            },
+            {
+                mqtt.CONF_USERNAME: "username",
+                mqtt.CONF_PASSWORD: "newpassword",
+            },
+            "newpassword",
+        ),
+        (
+            {
+                mqtt.CONF_BROKER: "test-broker",
+                mqtt.CONF_USERNAME: "username",
+                mqtt.CONF_PASSWORD: "verysecret",
+            },
+            {
+                mqtt.CONF_USERNAME: "username",
+                mqtt.CONF_PASSWORD: PWD_NOT_CHANGED,
+            },
+            "verysecret",
+        ),
+    ],
+)
+async def test_step_reauth(
+    hass: HomeAssistant,
+    mqtt_mock_entry: MqttMockHAClientGenerator,
+    mqtt_client_mock: MqttMockPahoClient,
+    mock_try_connection: MagicMock,
+    mock_reload_after_entry_update: MagicMock,
+    test_input: dict[str, Any],
+    user_input: dict[str, Any],
+    new_password: str,
+) -> None:
+    """Test that the reauth step works."""
+
+    # Prepare the config entry
+    config_entry = hass.config_entries.async_entries(mqtt.DOMAIN)[0]
+    hass.config_entries.async_update_entry(
+        config_entry,
+        data=test_input,
+    )
+    await mqtt_mock_entry()
+
+    # Show the form
+    result = await hass.config_entries.flow.async_init(
```

Comment:
I think its to call `entry.async_start_reauth(hass)` here as its an exposed api

Reply:
See: https://github.com/home-assistant/core/pull/116011/files#diff-454ebc7d7d7827598c55e6611dd5af8a2409744ab7bf06c3c4923867b8d7788fR1114-R1115